source/common/network/matching/inputs.h

```diff
@@ -35,6 +37,21 @@ class DestinationIPInput : public Matcher::DataInput<MatchingDataType> {
   Matcher::DataInputGetResult get(const MatchingDataType& data) const override;
 };
 
+template <>
+Matcher::DataInputGetResult
+DestinationIPInput<UdpMatchingData>::get(const UdpMatchingData& data) const;
+
+template <class MatchingDataType>
+Matcher::DataInputGetResult
+DestinationIPInput<MatchingDataType>::get(const MatchingDataType& data) const {
```

Comment:
Yes, and we can take a step further. The template member implementation can be inlined. We can directly move the code to line 37.